tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -615,6 +643,22 @@ SpirvEvalInfo SPIRVEmitter::loadIfGLValue(const Expr *expr) {
   return info;
 }
 
+SpirvEvalInfo SPIRVEmitter::loadIfAliasVarRef(const Expr *expr) {
+  auto info = doExpr(expr);
+
```

Comment:
Overall I feel like the code below should be added at the bottom of ``doExpr`` and eliminate the need for ``loadIfAliasVarRef`` function. This should also eliminate the need for manually updating some ``doExpr`` calls to ``loadIfAliasVarRef`` and potentially missing some.

Reply:
Good suggestion.

I tried to apply the change on this commit, but it doesn't play well with `processAssignment()` and `loadIfGLValue()`. Assignment of buffers goes wrong. I'll investigate and try to apply the change after all other related changes.